src/Service.Tests/Unittests/ConfigFileWatcherUnitTests.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.IO.Abstractions;
+using Azure.DataApiBuilder.Config;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Configurations;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Azure.DataApiBuilder.Service.Tests.Unittests
+{
+    [TestClass]
+    public class ConfigFileWatcherUnitTests
+    {
+        /// <summary>
+        /// Use the file system to add a file that matches the format of our config file.
+        /// Use that to create the needed config loader and provider, and then get the
+        /// actual config object from the provider, instantiating our config file watcher
+        /// in the process. Modify that file we created originally and verify we reload
+        /// the config correctly for the desired RuntimeOptions.
+        /// </summary>
+        [TestMethod]
+        public void HotReloadConfigRestRuntimeOptions()
+        {
+            string initialRestPath = "/api";
+            string updatedRestPath = "/rest";
+            string initialGQLPath = "/api";
+            string updatedGQLPath = "/gql";
+
+            bool initialRestEnabled = true;
+            bool updatedRestEnabled = false;
+            bool initialGQLEnabled = true;
+            bool updatedGQLEnabled = false;
+
+            bool initialGQLIntrospection = true;
+            bool updatedGQLIntrospection = false;
+
+            HostMode initialMode = HostMode.Development;
+            HostMode updatedMode = HostMode.Production;
+
+            string initialConfig = @"
+{
+  ""$schema"": ""https://github.com/Azure/data-api-builder/releases/download/vmajor.minor.patch/dab.draft.schema.json"",
+  ""data-source"": {
+    ""database-type"": ""mssql"",
+    ""connection-string"": ""Server=test;Database=test;User ID=test;Password=test;"",
```

Comment:
Provide `placeholder` as the password value if it absolutely needs to be set

Reply:
ill simply remove